electrum/lnpeer.py

```diff
@@ -2795,3 +2795,7 @@ def process_onion_packet(
         if self.network.config.TEST_FAIL_HTLCS_WITH_TEMP_NODE_FAILURE:
             raise OnionRoutingFailure(code=OnionFailureCode.TEMPORARY_NODE_FAILURE, data=b'')
         return processed_onion
+
+    def on_onion_message(self, payload):
+        if getattr(self.lnworker, 'onion_message_manager'):  # only on LNWallet
```

Comment:
I don't think this works. Maybe with `hasattr`, otherwise this probably raises AttributeError.

Looks like in some other places we are using `if self.lnworker == self.lnworker.network.lngossip` (negation would be needed here), or we could use `isinstance(self.lnworker, LNWallet)` (though LNWallet cannot be a top-level import in this file). All these options are somewhat ugly, but maybe they should be consistent at least -- we could factor it out into a method of lnpeer.

Reply:
> I don't think this works. Maybe with `hasattr`, otherwise this probably raises AttributeError.

Yep indeed. For now replaced with `hasattr`, as the other approaches are at least equally unelegant :)